tests/e2e/test_rosacli_cluster.go

```diff
@@ -854,6 +855,126 @@ var _ = Describe("Classic cluster creation validation",
+				rosalCommand = config.GenerateCommand(command)
+				for additionalSecurityGroupFlag := range securityGroups {
+					rosalCommand.DeleteFlag(additionalSecurityGroupFlag, true)
+				}
+
+				if profile.ClusterConfig.HCP {
+					for additionalSecurityGroupFlag, value := range securityGroups {
+						rosalCommand.AddFlags("--dry-run",
+							additionalSecurityGroupFlag, value,
+							"--version", "4.15.11",
+							"--hosted-cp", "-y")
+						output, err := rosaClient.Runner.RunCMD(strings.Split(rosalCommand.GetFullCommand(), " "))
+						Expect(err).To(HaveOccurred())
+						index = strings.Index(additionalSecurityGroupFlag, "a")
+						flagName = additionalSecurityGroupFlag[index:]
+						Expect(output.String()).To(ContainSubstring("Using '301721915996' as billing account"))
+						Expect(output.String()).To(ContainSubstring("add --billing-account xxxxxxxxxx to previous command"))
+						Expect(output.String()).To(ContainSubstring("Hosted clusters deprecate the --multi-az flag"))
+						Expect(output.String()).To(ContainSubstring(
+							"Parameter '%s' is not supported for Hosted Control Plane clusters",
+							flagName))
+						rosalCommand.DeleteFlag(additionalSecurityGroupFlag, true)
+					}
+					Skip("Security groups not supported from Hosted Control Plane cluster")
+				}
+
+				if profile.ClusterConfig.BYOVPC {
+					By("Try creating cluster with additional security groups and ocp version lower than 4.14")
+					for additionalSecurityGroupFlag, value := range securityGroups {
+						levelMap := map[string]string{
+							additionalSecurityGroupFlag: value,
+							"--version":                 ocpVersionBelow4_14,
+						}
+
+						rosalCommand.AddFlags(
+							"--dry-run",
+							additionalSecurityGroupFlag, levelMap[additionalSecurityGroupFlag],
+							"--version", ocpVersionBelow4_14, "-y")
+						output, err := rosaClient.Runner.RunCMD(strings.Split(rosalCommand.GetFullCommand(), " "))
+						Expect(err).To(HaveOccurred())
+						index = strings.Index(additionalSecurityGroupFlag, "a")
+						flagName = additionalSecurityGroupFlag[index:]
+						Expect(output.String()).To(ContainSubstring(
+							"Parameter '%s' is not supported prior to version '4.14.0'",
+							flagName))
+						rosalCommand.DeleteFlag(additionalSecurityGroupFlag, true)
+					}
+
+					By("Try creating cluster with invalid additional security groups")
+					for additionalSecurityGroupFlag, value := range invalidSecurityGroups {
+						rosalCommand.AddFlags("--dry-run", additionalSecurityGroupFlag, "invalid", "--version", ocpVersion4_14, "-y")
+						output, err := rosaClient.Runner.RunCMD(strings.Split(rosalCommand.GetFullCommand(), " "))
+						Expect(err).To(HaveOccurred())
+						Expect(output.String()).To(ContainSubstring("Creating cluster"))
+						Expect(output.String()).To(ContainSubstring(
+							"To view a list of clusters and their status, run 'rosa list clusters'"))
```

Comment:
out of curiosity, why do we care about this substring existing in this particular output?

Reply:
@OAharoni-RedHat i see its a good chance to clarify the doubt in my mind, i was under the impression that if say there are 3 lines in ouput and we want to match the 3rd one, so can we directly match the 3rd without thinking about the earlier two lines?

Reply:
Hmm I think I may have not been clear. I don't mind that we're checking substrings in general, just that I'm not sure how checking to see if `To view a list of clusters and their status, run 'rosa list clusters'` coming up in the substring helps us reach the goal of this test. It seems to me like we just want to make sure that the 3 line here which checks for `Security Group ID '%s' doesn't have 'sg-' prefix` is the only necessary substring we want to find. I recognize this is a bit of a nitpick, so feel free to ignore it!